### PR TITLE
Use github action to release nuget package and binaries

### PR DESCRIPTION
**Pull request by 928PJY**
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/docfx/pull/7098)

https://github.com/dotnet/docfx/runs/1985812819?check_suite_focus=true
Release succeeded and confirmed github release and nuget package published.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  push:
+    branches: [ v3-release ]
+jobs:
+  build:
+    name: Release
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (mingwli) [inline .github/workflows/release.yml:12]: I guess that depth 0 will make the generated version number `{height}` `1`?

Comment (yufeih) [reply .github/workflows/release.yml:12]: `fetch-depth`: 0 means fetch all commits

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Create GtiHub release
+      uses: marvinpinto/action-automatic-releases@latest
+      with:
+        repo_token: "${{ secrets.GITHUB_TOKEN }}"
+        automatic_release_tag: ${{ env.GitAssemblyInformationalVersion }}
+        title: Version ${{ env.GitAssemblyInformationalVersion }}
+        prerelease: true
+        files: |
+          drop/docfx-bin/staging/**/*
+    - name: Set up nuget
```
Comment (yufeih) [inline .github/workflows/release.yml:29]: No need to install NuGet, use `dotnet` to publish
```suggestion
    - run: dotnet nuget push --api-key "" --source "" --skip-duplicate
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        fetch-depth: 0
+    - uses: actions/setup-dotnet@v1
+    - name: Build
+      shell: pwsh
+      run: ./build.ps1
+      id: build-binaries
+    - name: Test
+      run: dotnet test -c Release
+    - name: Create GtiHub release
+      uses: marvinpinto/action-automatic-releases@latest
```
Comment (yufeih) [inline .github/workflows/release.yml:21]: Update `README.md` _Binary Builds` section to link to the preview releases page.

File: build.ps1 [hunk 4]
```diff
@@ -30,8 +30,6 @@ function publishBinaryPackages() {
         $packageName = "docfx-$rid-$version"
         Compress-Archive -Path "$packagesBasePath/$rid/*" -DestinationPath "$stagingPath/$packageName.zip" -Update
         New-Item -Path "$stagingPath" -Name "$packageName.zip.sha256" -Force -ItemType "file" -Value (Get-FileHash "$stagingPath/$packageName.zip").Hash
-        Copy-Item "$stagingPath/$packageName.zip" "$stagingPath/docfx-$rid-latest.zip" 
-        Copy-Item "$stagingPath/$packageName.zip.sha256" "$stagingPath/docfx-$rid-latest.zip.sha256"
```
Comment (yufeih) [inline build.ps1:34]: consider move `dotnet pack` (in `publish` function) to `release.yml`, and move the `publishBinaryPackages` function to a different step in `release.yml` then remove `build.ps1`

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      with:
+        repo_token: "${{ secrets.GITHUB_TOKEN }}"
+        automatic_release_tag: ${{ env.GitAssemblyInformationalVersion }}
```
[the remark this answers is not in the document]
Comment (yufeih) [reply .github/workflows/release.yml:24]: I see it is autopopulated by NerdBank.GitVersioning

File: .github/workflows/release.yml (hunk 3 above)
Comment (shanluo911) [reply .github/workflows/release.yml:21]: added

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-dotnet@v1
+    - name: Donet pack
+      # if: ${{ env.BUILD_REASON != 'pull_request' }}
+      shell: pwsh
+      run: |
```
Comment (yufeih) [inline .github/workflows/release.yml:17]: Remove the `Remove-Item` as we are on a hosted machine.
```suggestion
      run: dotnet pack -c Release -o drop
```

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+jobs:
+  build:
+    name: Release
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-dotnet@v1
+    - name: Donet pack
```
Comment (yufeih) [inline .github/workflows/release.yml:14]: nit: remove `name`, `shell`, `-run: xyz` is enough

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+          if ($rid -eq "win7-x64") {
+            $version = Invoke-Expression "$packagesBasePath/win7-x64/docfx.exe --version"
+            Write-Host "package version: $version"
+          }
+          $packageName = "docfx-$rid-$version"
+          Compress-Archive -Path "$packagesBasePath/$rid/*" -DestinationPath "$stagingPath/$packageName.zip" -Update
+          New-Item -Path "$stagingPath" -Name "$packageName.zip.sha256" -Force -ItemType "file" -Value (Get-FileHash "$stagingPath/$packageName.zip").Hash
+        }
+    - name: Test
+      run: dotnet test -c Release
```
Comment (yufeih) [inline .github/workflows/release.yml:37]: run test before pack and build binary package

**Review by yufeih — APPROVED** [review]

**Pull request merged**